### PR TITLE
fix: Runtime Validation in CreateCopyright BaseFragment

### LINKED ISSUE #712: Providing runtime validation for fields in create copyright form
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    

**Actual Behaviour**

When any incorrect url is entered in the url field and the form is submitted the server gives `HTTP 422 UNPROCESSABLE ENTITY`.

**Expected Behaviour**

There should be runtime validation of text input fields so that user can get feedback for incorrect input before submitting the form.

**Steps to reproduce it**

Enter incorrect url or year and then submite the form.

**LogCat for the issue**

`E/Logger: An exception occurred : HTTP 422 UNPROCESSABLE ENTITY
          retrofit2.adapter.rxjava2.HttpException: HTTP 422 UNPROCESSABLE ENTITY
              at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:54)
              at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
              at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:43)
              at io.reactivex.Observable.subscribe(Observable.java:11442)
              at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
              at io.reactivex.Observable.subscribe(Observable.java:11442)
              at io.reactivex.internal.operators.observable.ObservableDoOnEach.subscribeActual(ObservableDoOnEach.java:42)
              at io.reactivex.Observable.subscribe(Observable.java:11442)
              at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
              at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:571)
              at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
              at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
              at java.util.concurrent.FutureTask.run(FutureTask.java:237)
              at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
              at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
              at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
              at java.lang.Thread.run(Thread.java:761)`

**Android version and Phone Model**

Moto G4 Plus
   

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #712 

Changes: 
1. Changed AboutEventActivity to support all fragments(CreateCopyright, UpdateCopyright) from same about activity fragment container.
2. Added Runtime Validation of fields in CreateCopyrightFragment.
3. Added Validation for Year field on clicking submit. 
4. Changed CreateCopyright Fragment from BaseBottomSheetFragment to BaseFragment.
5. Changed UpdateCopyrightFragment from BaseBottomSheetFragment to BaseFragment.

Screenshots for the change: 
20180330192802-7c011ad592 gif-2-mp4 com
20180330193450-7c011ad592 gif-2-mp4 com

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/core/event/copyright/ICreateCopyrightView.java [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+
+import org.fossasia.openevent.app.common.Function;
 import org.fossasia.openevent.app.common.mvp.view.Erroneous;
 import org.fossasia.openevent.app.common.mvp.view.Progressive;
 import org.fossasia.openevent.app.common.mvp.view.Successful;
 
+
 public interface ICreateCopyrightView extends Progressive, Erroneous, Successful {
 
     void dismiss();
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/event/copyright/ICreateCopyrightView.java]: Remove this

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/core/event/copyright/ICreateCopyrightView.java]: Or move back to about event once done

File: app/src/main/java/org/fossasia/openevent/app/utils/ValidateUtils.java [hunk 2]
```diff
@@ -11,6 +11,8 @@
     private static final Pattern VALID_URL_REGEX =
         Pattern.compile("^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]");
 
+    private static final Pattern VALID_YEAR_REGEX = Pattern.compile("^\\d{4}$");
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/utils/ValidateUtils.java]: Regexes are slow. This is a fairly trivial check. Can be done via static checking

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/utils/ValidateUtils.java]: I think databinding validation will work as well

**Comment by iamareebjamal** [comment]
Also, you have a PMD violation

https://141-65606855-gh.circle-artifacts.com/0/outputs/pmd/pmd.html

**Comment by Masquerade0097** [comment]
@iamareebjamal Changes done. Please review.

**Pull request merged**